vision_datasets/common/data_manifest.py

```diff
@@ -96,14 +98,28 @@ def __init__(self, id, img_path, width, height, labels):
                 classification: [c_id] for multiclass, [c_id1, c_id2, ...] for multilabel;
                 detection: [[c_id, left, top, right, bottom], ...];
                 image caption: [caption1, caption2, ...];
-                image_text_matching: [(text1, match (0 or 1), text2, match (0 or 1), ...)]
+                image_text_matching: [(text1, match (0 or 1), text2, match (0 or 1), ...)];
+                image matting: [image_matting], where 'image_matting' is a PIL object or a Numpy array;
                 multitask: dict[task, labels]
+            label_file_path (str): path to label file
         """
         self.id = id
         self.img_path = img_path
         self.width = width
         self.height = height
-        self.labels = labels
+        self._labels = labels
+        self.label_file_path = label_file_path
+
+    @property
+    def labels(self):
+        if self.label_file_path:
```

Comment:
Image.open(FileReader().open(self.label_file_path)) returns an Image, while self._labels returns a list of images